crypto/hrss/hrss.c

```diff
@@ -914,20 +913,28 @@ void HRSS_poly3_invert(struct poly3 *out, const struct poly3 *in) {
 // Coefficients are ordered little-endian, thus the coefficient of x^0 is the
 // first element of the array.
 struct poly {
-#if defined(HRSS_HAVE_VECTOR_UNIT)
-  union {
-    // N + 3 = 704, which is a multiple of 64 and thus aligns things, esp for
-    // the vector code.
-    uint16_t v[N + 3];
-    vec_t vectors[VECS_PER_POLY];
-  };
```

Comment:
why not just name the union here instead of creating the new struct?

Reply:
The change scope will be increased a lot because the previous `poly` has two `v` (one is inside `union`, another one is outside `union`), and both `v` are accessed by `p.v`. If adding the name to the union, I may need to track down which `v` access should be modified.

Credit to [Nir's code](https://sim.amazon.com/issues/CryptoAlg-626?selectedConversation=efe9c26a-a934-4c4f-9c10-51293efd89eb).

Reply:
Reverted. I will not make any fix on c99 related warnings.